internal/messaging/identification.go

```diff
@@ -0,0 +1,131 @@
+package messaging
+
+import (
+	"log"
+	"math/big"
+
+	config "github.com/chain4travel/camino-messenger-bot/config"
+	"github.com/chain4travel/camino-messenger-contracts/go/contracts/cmaccount"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"go.uber.org/zap"
+	"maunium.net/go/mautrix/id"
+)
+
+var _ IdentificationHandler = (*evmIdentificationHandler)(nil)
+
+var roleHash = crypto.Keccak256Hash([]byte("CHEQUE_OPERATOR_ROLE"))
+
+type evmIdentificationHandler struct {
+	ethClient       *ethclient.Client
+	cfg             *config.EvmConfig
+	matrixHost      string
+	cmAccountBotMap map[common.Address]id.UserID
+}
+
+type IdentificationHandler interface {
+	getAllBotAddressesFromCMAccountAddress(cmAccountAddress common.Address) ([]string, error)
+	getFirstBotFromCMAccountAddress(cmAccountAddress common.Address) (string, error)
+	isMyCMAccount(cmAccountAddress common.Address) bool
+	getMyCMAccountAddress() common.Address
+	getMatrixHost() string
+	isBotInCMAccount(common.Address, common.Address) (bool, error)
+	getCmAccount(id.UserID) (common.Address, bool)
+	addToMap(common.Address, id.UserID)
+	getBotFromMap(common.Address) (bool, id.UserID)
+}
+
+func NewIdentificationHandler(ethClient *ethclient.Client, _ *zap.SugaredLogger, cfg *config.EvmConfig, mCfg *config.MatrixConfig) (IdentificationHandler, error) {
+	return &evmIdentificationHandler{
+		ethClient:       ethClient,
+		cfg:             cfg,
+		matrixHost:      mCfg.Host,
+		cmAccountBotMap: make(map[common.Address]id.UserID),
```

Comment:
you need to add rw mutex and do r/w locks for read/write on this map, since it can be accessed concurently